components/org.wso2.carbon.identity.organization.resource.sharing.policy.management/src/main/java/org/wso2/carbon/identity/organization/resource/sharing/policy/management/constant/PolicyEnum.java

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.organization.resource.sharing.policy.management.constant;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.wso2.carbon.identity.organization.resource.sharing.policy.management.constant.ResourceSharingConstants.APPLICATION;
+import static org.wso2.carbon.identity.organization.resource.sharing.policy.management.constant.ResourceSharingConstants.IDP;
+import static org.wso2.carbon.identity.organization.resource.sharing.policy.management.constant.ResourceSharingConstants.USER;
+
+/**
+ * Enum representing user sharing policies with additional fields for code, name, applicable resources, and details.
+ */
+public enum PolicyEnum {
+
+    ALL_EXISTING_ORGS_ONLY(
+            "GEN-EO-0001",
+            "AllExistingOrgsOnly",
+            "ALL_EXISTING_ORGS_ONLY",
+            Arrays.asList(USER, APPLICATION, IDP),
+            "This policy applies when the resource needs to be shared with all existing organizations at " +
+                    "the current time. Newly created organizations after the policy is applied will not be included " +
+                    "under this policy."),
+    ALL_EXISTING_AND_FUTURE_ORGS(
+            "GEN-EF-0002",
```

Comment:
```suggestion
            "GEN-EFO-0002",
```
Lets represent **E**xisting and **F**uture **O**rgs

Fix all other affecting policies